src/track/serato/tags.cpp

```diff
@@ -20,10 +20,10 @@
 namespace {
 
 QString getPrimaryDecoderNameForFilePath(const QString& filePath) {
-    const QString fileExtension =
+    const QString fileType =
             mixxx::SoundSource::getTypeFromFile(QFileInfo(filePath));
```

Comment:
I think SoundSource::getTypeFromFile() does not fit to the rest of the code now. We can't be sure that the underlying magic number libraries returns the file preferedSuffix where we have a provider for. We don't have it under our control. I can change at any time when it is updated. I think we need to return all file suffixes and probe all to be sure.

Reply:
I have no idea what you ask from me.

Reply:
SoundSource::getTypeFromFile does the flowing:

* Look up the magic number in the file content 
* pick the mime type matching to this magic number 
* Return 1 of many suffixes which is considered as preferred.
* Return this preferred suffix as file type, even if we don't have such file type, and even if the original suffix was already matching a file type. 

This works only by a chance for all our sound sources and may break at any time. I have most concerned with ogg an mod.

A workaround would be to return all suffixed for this magic number and look, if one of them is a fileType. 

To secure this against regression that might happen due to an update of the magic numbers database, we should test these cases as a unit-test. Or verify that this is already covered by a unit-test.  

A much better workaround would be to replace the  QString fileTypeg by a MimeType value.

Reply:
I am not going to replace the legacy string types, out of scope of this PR.

Reply:
https://github.com/mixxxdj/mixxx/pull/4608/commits/e40fab905e5d3a84e3f9c63bfbda924f90e0545f

Reply:
Please reopen if you need more tests or otherwise provide those additional tests that you want to see.

Reply:
The issue pending issue is the following: 
* if a file with a magic number is for instance a "Ogg Vorbis Codec compressed Multimedia file"
* Possible extensions are "OGA, OGG, OGV, OGX"
* Qt returns one of them, based on an underlying system library not under our control. 
* By luck it will return OGG, But it might one of the others. 
* Result OGG files are not playbable anymore. This can happen due to an OS update. 

I consider this as a rock solid solution: 

Fetch all extensions from QT: Which are "OGA, OGG, OGV, OGX" and check them one by one for a sound-source. 
So the second Extension will a match and the problem is solved. 

We also need to makes sure that the FileType is considered as OGA and not OGG. 

An alternative solution would be to maintain our own list of preferred extensions for each MimeType.

Reply:
Does this PR introduce a regression to the current state? If yes then please clarify how. If not then please merge it as.

I am not interested in extending the scope of this PR. It was only intended to fix the inconsistencies, not to improve the questionable *file type* concept that should better be replaced entirely by MIME types.

Reply:
If there is a regression then please provide a test that demonstrates it.

Reply:
The type "ogg" will always map to "ogg" if this is one of the file suffixes provided by QMimeDatabase. I have even written a test that verifies this assumption!

We cannot test all possible situations for files with wrong suffixes. Use at you own risk or fix your file suffixes. I am done here.

Reply:
This PR scans for now for these files: 
"*.3g2", "*.3gp2", "*.3gpp2", "*.3gp", "*.3gpp", "*.3ga", "*.aac", "*.adts", "*.ass", "*.aiff", "*.aif", "*.flac", "*.it", "*.m4a", "*.f4a", "*.mp4", "*.m4v", "*.f4v", "*.lrv", "*.mod", "*.ult", "*.uni", "*.m15", "*.mtm", "*.669", "*.med", "*.mj2", "*.mjp2", "*.qt", "*.mov", "*.moov", "*.qtvr", "*.mp3", "*.mpga", "*.oga", "*.ogg", "*.opus", "*.ogv", "*.s3m", "*.stm", "*.wav", "*.wv", "*.wvp", "*.xm"

before it was 
"*.3g2", "*.3gp", "*.aac", "*.aif", "*.aiff", "*.caf", "*.flac", "*.it", "*.m4a", "*.m4v", "*.med", "*.mj2", "*.mod", "*.mov", "*.mp3", "*.mp4", "*.ogg", "*.okt", "*.opus", "*.s3m", "*.stm", "*.wav", "*.wv", "*.xm"

Now for instance a "*.3gpp2" file or an "*.oga" is listed in the library, but it is not playable. My guess is, that all new extensions are affected. 

If you do not have interest to solve this yourself, I can offer to issue a PR against your branch. But let's discuss fist which solution will be acceptable. I see the options already outlined above: 
* Probe all sibiling suffixes when selecting the soundsource 
* Maintain a lookup table with the correct FileType string per mimeType 
* Replace FileType by QMimeTime. 

What do you think?

Reply:
No. Either merge this PR or prove me wrong with a regression test that previously worked.

Reply:
I will not do the effort of writing a test. You can easily test it manually. Just rename a ogg file to oga. Rescan your library, the new fille will show up with FilyType oga. Now try to play it -> does not work. 

This is not a regression, but an incomplete feature of this branch, I consider as a bug. 

But anyway, I am happy to help. Did you consider the three options for a solution? Which one do you like to merge?

Reply:
If it's not a regression, it must not block merging this PR!
Because this PR fixes other issue and doesn't introduce any new bug. Any further improvements can be done in follow-up PRs.

Reply:
@JoergAtGithub Thank you!

Hardly anyone will be affected by the remaining shortcomings and I refuse to spend more time on this.

Reply:
Which bug is fixed by this PR? 

From the user perspective it finds new files when scanning the library, but they cannot be played. The bad thing in this is that, these new files are persistent in the database. So we must not merge this unless we are willing to provide a clean up routine in a follow up PR.

But back to topic. I have already offered to bring this in a merge-able state. So let's look into the possibles solutions? Which one will fit?  I need that info because I have no interest to do one version and change it to another in a code review.